File: python/p4.py
def calc():
    for n in (x for x in xrange(999*999, 100*100, -1) if str(x) == str(x)[::-1]):
        for i in xrange(999, 99, -1):
            if n % i == 0 and 999 >= n / i >= 100:
                return n

print calc()

File: go/p4.go
package main

import (
    "fmt"
    "strconv"
)

func reverse_string(s string) string {
    runes := []rune(s)
    for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
        runes[i], runes[j] = runes[j], runes[i]
    }
    return string(runes)
}

func calc() int {
    for n := 999*999; n >= 100*100; n-- {
        s := strconv.Itoa(n)
        if s != reverse_string(s) { continue }
        for i := 999; i >= 99; i-- {
            if n % i != 0 { continue }
            d := n / i;
            if d <= 999 && d >= 100 {
                return n;
            }
        }
    }
    return 0
}

func main() {
    rslt := calc()
    fmt.Println(rslt)
}
